homeassistant/components/aqara/__init__.py

```diff
@@ -0,0 +1,196 @@
+"""Support for Aqara Smart devices."""
+from __future__ import annotations
+
+import logging
+from typing import NamedTuple
+
+from aqara_iot import (
+    AqaraDeviceListener,
+    AqaraDeviceManager,
+    AqaraHomeManager,
+    AqaraOpenAPI,
+    AqaraOpenMQ,
+    AqaraPoint,
+)
+import requests
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.dispatcher import async_dispatcher_connect, dispatcher_send
+
+from .const import (
+    AQARA_DISCOVERY_NEW,
+    AQARA_HA_SIGNAL_REGISTER_POINT,
+    AQARA_HA_SIGNAL_UPDATE_ENTITY,
+    AQARA_HA_SIGNAL_UPDATE_POINT_VALUE,
+    CONF_COUNTRY_CODE,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    DOMAIN,
+    PLATFORMS,
+)
+from .util import string_dot_to_underline, string_underline_to_dot
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class HomeAssistantAqaraData(NamedTuple):
```

Comment:
why not a `dataclass` here? That's been the standard used in the core integrations instead of `NamedTuple`

Reply:
As this is a fixed set of data, this is actually OK. This uses less memory than dataclass.